tests/csm/rest/test_iam_users.py

```diff
@@ -424,24 +437,234 @@ def test_35930(self):
+                system_utils.execute_cmd(cmd=common_cmd.CMD_MKDIR.format(TEST_DATA_FOLDER))
+            system_utils.create_file(file_path_upload, self.file_size)
+            resp = s3_obj.put_object(bucket_name=self.bucket_name, object_name=test_file,
+                                     file_path=file_path_upload)
+            self.log.info("Removing uploaded object from a local path.")
+            os.remove(file_path_upload)
+            assert_utils.assert_true(resp[0], resp[1])
+            self.log.info("Step: Verify get object.")
+            resp = s3_obj.get_object(self.bucket_name, test_file)
+            assert_utils.assert_true(resp[0], resp)
+        self.log.info("[END]Creating IAM users with different tenant")
+        self.log.info("##### Test completed -  %s #####", test_case_name)
+
+    # pylint: disable=too-many-statements
+    @pytest.mark.lc
+    @pytest.mark.csmrest
+    @pytest.mark.s3_ops
+    @pytest.mark.tags('TEST-39411')
+    def test_39411(self):
+        """
+        S3 IAM User: Create same name buckets with IAM users of same name in different tenant
+        """
+        test_case_name = cortxlogging.get_frame()
+        self.log.info("##### Test started -  %s #####", test_case_name)
+        self.log.info(
+            "[START] Create same name buckets with same name IAM users in different tenant")
+        for cnt in range(2):
+            tenant = "tenant_" + str(cnt)
+            self.log.info("Creating new iam user with tenant %s", tenant)
+            optional_payload = self.csm_obj.iam_user_payload_rgw("loaded")
+            optional_payload.update({"tenant": tenant})
+            optional_payload.update({"display_name": self.display_name})
+            self.log.info("updated payload :  %s", optional_payload)
+            resp1 = self.csm_obj.create_iam_user_rgw(optional_payload)
+            self.log.info("Verify Response : %s", resp1)
+            assert_utils.assert_true(resp1.status_code == HTTPStatus.CREATED,
+                                     "IAM user creation failed")
+            self.created_iam_users.add(resp1.json()['tenant'] + "$" + optional_payload['uid'])
+            resp = self.csm_obj.compare_iam_payload_response(resp1, optional_payload)
+            self.log.info("Printing response %s", resp)
+            assert_utils.assert_true(resp[0], resp[1])
+            self.log.info("Step: Verify no bucket present in new account")
+            s3_obj = S3TestLib(access_key=resp1.json()["keys"][0]["access_key"],
+                               secret_key=resp1.json()["keys"][0]["secret_key"])
+            buckets = s3_obj.bucket_list()[1]
+            assert_utils.assert_false(len(buckets), "buckets found on new IAM user")
+            self.log.info("Step: Verified no bucket present in new account")
+            self.log.info("Create bucket and perform IO")
+            self.log.info("Step: Verify create bucket")
+            status, resp = s3_obj.create_bucket(self.bucket_name)
```

Comment:
Lines 546 - 570 can be replaced by create_bucket_put_object function from s3_common_test_lib.

Reply:
done